sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -0,0 +1,198 @@
+            ).to_string()
+        )
+        return ans
+
+    def propose_spdx_id(self) -> str:
+        purl_hex_digest = hashlib.sha256(self.purls()[0].encode()).hexdigest()
+        return f"SPDXRef-Image-{self.name}-{purl_hex_digest}"
+
+
+def create_package(image: Image, spdxid: Optional[str] = None, image_index_digest: Optional[str] = None) -> dict:
+    return {
+        "SPDXID": image.propose_spdx_id() if not spdxid else spdxid,
+        "name": image.name if not image.arch else f"{image.name}_{image.arch}",
+        "versionInfo": image.tag,
+        "supplier": "Organization: Red Hat",
+        "downloadLocation": "NOASSERTION",
+        "licenseDeclared": "NOASSERTION",
+        "externalRefs": [
+            {
+                "referenceCategory": "PACKAGE-MANAGER",
+                "referenceType": "purl",
+                "referenceLocator": purl,
+            }
+            for purl in image.purls(image_index_digest)
+        ],
+        "checksums": [
+            {
+                "algorithm": image.digest_algo,
+                "checksumValue": image.digest_hex_val,
+            }
+        ],
+    }
+
+
+def get_relationship(spdxid: str, related_spdxid: str):
+    return {
+        "spdxElementId": spdxid,
+        "relationshipType": "VARIANT_OF",
+        "relatedSpdxElement": related_spdxid,
+    }
+
+
+def create_sbom(
+    image_index_url: str,
+    image_index_digest: str,
+    inspect_input: dict[str, Any],
+) -> dict:
+    if inspect_input["mediaType"] != "application/vnd.oci.image.index.v1+json":
+        raise ValueError("Invalid input file detected, requires `buildah manifest inspect` json.")
+
+    image_index_obj = Image.from_image_index_url_and_digest(image_index_url, image_index_digest)
+    sbom_name = f"{image_index_obj.name}-{image_index_obj.tag}"
+
+    packages = [create_package(image_index_obj, "SPDXRef-image-index")]
+    relationships = [
+        {
+            "spdxElementId": "SPDXRef-DOCUMENT",
+            "relationshipType": "DESCRIBES",
+            "relatedSpdxElement": "SPDXRef-image-index",
+        }
+    ]
+
+    for manifest in inspect_input["manifests"]:
+        if manifest["mediaType"] != "application/vnd.oci.image.manifest.v1+json":
+            continue
+
+        arch_image = Image(
+            arch=manifest.get("platform", {}).get("architecture"),
+            name=image_index_obj.name,
+            digest=manifest.get("digest"),
+            tag=image_index_obj.tag,
+            repository=image_index_obj.repository,
+        )
+        packages.append(create_package(arch_image, image_index_digest=image_index_obj.digest))
+        relationships.append(get_relationship(arch_image.propose_spdx_id(), "SPDXRef-image-index"))
+
+    sbom = {
+        "spdxVersion": "SPDX-2.3",
+        "dataLicense": "CC0-1.0",
+        "documentNamespace": f"https://redhat.com/spdxdocs/{sbom_name}-{uuid4()}",
```

Comment:
Similar concern about hardcoding Red Hat here

Reply:
Changed `redhat.com` to `konflux-ci.dev`.

Reply:
Oh, it's just a "made up" URL. This works :smile: